### ISSUE TITLE
please highlight non-ignored fields

### ISSUE THREAD

**Issue by ckhung** [grafana/grafana-csv-datasource#168]
Can we highlight or somehow mark the non-ignored fields (say in the header) please? I have setup a simple csv file [satellites.csv](https://www.cyut.edu.tw/~ckhung/saas/scatplot/satellites.csv) as the datasource. I then add 3 fields name, orbit_major, and rev_cycle as the non-ignored fields. When I click on the header of orbit_major, however, it is still sorted as strings instead of (desired) as numbers. (See screenshot.) Same with rev_cycle. In fact, when I click "ignore unknown", only name appears. Apparently the system does not recognize the two numerical fields. Yet there is not error message or any hint. Is there something else I have to click after adding a field? It does not help even if I click the big blue "run query" button.

If I may complain a bit... Grafana seems very powerful but unfortunately it does not provide an intuitive interface for simple tasks. As a linux power user and fan of ["simple things simple, complex things possible"](https://news.ycombinator.com/item?id=3329675)  but complete newbie to grafana, I don't want to begin with the heavy-weight [tutorial-environment](https://github.com/grafana/tutorial-environment) which creates several dockers even before I can do anything. I want to begin with simple a csv. I am grateful for the grafana-csv-datasource plugin. (Why is it not a core functionality of grafana?) This project could be a very helpful on-ramp for people who share my preferences. I surely would like to write a tutorial to help people avoid the hours of frustration I have gone through. Regular expressions are so much easier and intuitive  compared to grafana!

I am running grafana-enterprise 8.4.2 on linux mint 20.2 "uma", xfce variant.

![1](https://user-images.githubusercontent.com/3609806/155888094-677853ef-7d80-4c2d-94a2-0f280b1b2ff4.png)

**Comment by marcusolsson** [comment]
Hm. Looking at your schema, it should indeed have been interpreted as a number. Could you perhaps share your (anonymized) CSV data so that I can replicate it?

---

Regarding your feedback on the tutorial, Grafana has (is?) primary been a tool for visualizing dynamic time series data such as network traffic or CPU load in near real-time, rather than for more static analytics purposes. I think this also partially answers why there isn't any built-in support for CSV. 

Recent versions of Grafana have introduced more visualizations and features for business intelligence and analytics purposes, but I suspect systems monitoring will continue to be the primary use case for Grafana for some time to come.

**Comment by ckhung** [comment]
This is the CSV data file: [satellites.csv](https://www.cyut.edu.tw/~ckhung/saas/scatplot/satellites.csv) I figured out that the cause is those extra spaces in my data file. The problem goes away if I use the space-free version instead [sat.csv](https://www.cyut.edu.tw/~ckhung/saas/scatplot/sat.csv) 

Still, it would be greatly helpful if the plugin can give some error message such as ' "radius" is not a valid field name'. Lack of error message is the most frustrating part. Not only for users, but also for developers when you try to help newbies who make strange mistakes. Another more elegant solution would be making the "field" field as a dropdown list instead of a free-text entry box so that the user does not even have an opportunity of making meaningless mistakes.

Thanks for the explanation about time series data. I kind of guessed that was the case. For example I was surprised that bubble chart is not among the supported charts. Also I found it very counter-intuitive to see the three numerical fields to be aggregated into a single histogram. Different numerical fields may not even have the same unit. How can it make sense to aggregate them? I expected the histogram only displaying one single numerical field, possibly marked with different colors according to another categorical field (for example planet to which the satellite belongs). But I digress...

Anyway I should probably look for some other open source visualization program for my static data need. In any case, many thanks for your great work. Hopefully some day I will come back to grafana. I still think that he csv plugin and the json plugin would be better starting point for newbies.
